src/components/ModalLedgerSignToken.js

```diff
@@ -0,0 +1,264 @@
+      setError('invalid_name');
+      return false;
+    }
+
+    return true;
+  }
+
+  // verify token information
+  const verifyToken = () => {
+    // if the token data is invalid, we don't need to verify the signature
+    if (!validateTokenData()) return;
+
+    const tokenSignatures = tokens.getTokenSignatures();
+    console.log('tokenSignatures', tokenSignatures);
+    if (tokenSignatures.hasOwnProperty(token.uid)) {
+      console.log("Wait for verification");
+      // the token has a signature on storage, we need to verify it
+      const signedToken = {
+        'uid': token.uid,
+        'name': token.name,
+        'symbol': token.symbol,
+        'signature': tokenSignatures[token.uid],
+      };
+      console.log('signedToken', signedToken);
+      setWaitingLedger(true);
+      ledger.verifyTokenSignature(signedToken);
+    } else {
+      console.log("No signature on storage.");
+      canSend(true);
+    }
+  }
+
+  /**
+   * Method called when user clicks the button to register the token  
+   * Validates that the data written is valid
+   *
+   * @param {Object} e Event emitted when user clicks the button
+   */
+  const handleSend = () => {
+    canSend(false);
+    setOk(false);
+    setError(null);
+    // send token data to ledger
+    setWaitingLedger(true);
+    ledger.signToken(token);
+  }
+
+  /**
+   * Handle the response of a send token data call to Ledger.
+   * Will list all tokens that failed verification, if all pass then start sign tx
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleSignature = (event, arg) => {
+    console.log('handleSignature', arg);
+    setWaitingLedger(false);
+    if (arg.success) {
+      // save token signature on storage
+      tokens.addTokenSignature(token.uid, arg.data.toString("hex"));
+      setOk(true);
+    } else {
+      switch (arg.error.status) {
+        // user deny
+        case 0x6985: 
+          setError('user_deny');
+          break;
+        case 0xb00a: 
+          setError('invalid_token');
+          break;
+        default:
+          setError('unknown_error');
+      }
+    }
+  }
+
+  /**
+   * Handle the response of a send token data call to Ledger.
+   * Will list all tokens that failed verification, if all pass then start sign tx
+   *
+   * @param {IpcRendererEvent} event May be used to reply to the event
+   * @param {Object} arg Data returned from the send token data call
+   */
+  const handleVerify = (event, arg) => {
+    console.log('handleVerify', arg);
+    setWaitingLedger(false);
+    if (arg.success) {
+      setOk(true);
+    } else {
+      // Invalid signature, we should start the signing process
+      canSend(true);
+    }
+  }
+
+  /**
+   * Effects
+   */
+
+  // useEffect second parameter [] ensures this runs only once
+  useEffect(() => {
+    $(`#${modalId}`).on('hide.bs.modal', (e) => {
```

Comment:
I prefer using `hidden.bs.modal` instead of `hide` here, otherwise the modal state data may 'blink' because it will be cleaned before the modal is fully hidden.

Reply:
Done